Fix warnings from pytest 3.8

tomjnixon opened the pull request:
This fixes #2.

I had to make a few other changes to fix up the build system, and added a test for some implicit behaviour of the existing code.

This version is not compatible with pytest versions before 3.6, which means it wont run on python 2.6. IMO this is a good thing, as python 2.6 is no longer receiving security patches. If you would rather keep compatibility then there's a commit that does that on my `misc_fixes_compat` branch.

omarkohl commented:
Hey, thanks for the effort! :-) I'll have a look as soon as I can.

tomjnixon commented on setup.py:
```diff
@@ -13,7 +13,7 @@ def _read(fname):
 DEPENDENCIES = [
     'py',
-    'pytest',
+    'pytest~=3.6',
```
It's documented here:

https://www.python.org/dev/peps/pep-0440/#compatible-release

and declares that it requires a pytest release that's compatible with 3.6 (i.e. before 4.x); would you prefer `pytest == 3.*, >= 3.6`?

omarkohl replied:
Thanks for the link.

I think >= 3.6 is best. In general I prefer to be permissive with the versions unless I know there is an issue. People should be encouraged not prevented from upgrading.

As a side-note: To ensure stable and reproducible environments I would always recommend exactly pinning your versions (e.g. using pip-tools or Pipenv) and trying to upgrade those versions in regular intervals.

tomjnixon commented on pytest_datafiles.py:
```diff
@@ -66,16 +66,20 @@ def datafiles(request, tmpdir):
     keep_top_dir = options['keep_top_dir']
+
+    if not content:
+        return tmpdir
```
Sure. I intended to separate the mark parsing and the logic, but either is fine by me.

omarkohl replied:
You did not fix this... but anyway it was not very important.

tomjnixon replied:
Sorry, I mis-interpreted the meaning of your thumbs-up. Thanks for merging anyway...

tomjnixon commented:
Hi, sorry about the delay. I've changed the version requirement as you suggested.

I also found and fixed a bug in the tests that was causing them to fail on one machine but not another; see the latest commit. I'd normally open another PR but there's already a bunch of unrelated commits in here...

Thanks!
